File: task1/GO/main.go
package main

import "fmt"

func allSymvl(str string, t string) bool {
	var count [128]int

	for i := 0; i < len(str); i++ {
		c := str[i]
		count[c]++
	}

	for i := 0; i < len(t); i++ {
		c := t[i]
		count[c]--
		if count[c] < 0 {
			return false
		}
	}
	return true
}

func minStr(s string, t string) string {
	if len(t) > len(s) {
		return "Не найдено"
	}
	result := ""
	minLen := len(s) + 1

	for i := 0; i < len(s); i++ {
		for j := i; j < len(s); j++ {
			str := s[i : j+1]

			if allSymvl(str, t) {
				if len(str) < minLen {
					minLen = len(str)
					result = str
				}
				break
			}
		}
	}
	if len(result) == 0 {
		return "Не найдено"
	}
	return result
}

func main() {
	var s, t string
	fmt.Scan(&s)
	fmt.Scan(&t)
	fmt.Println(minStr(s, t))
}

File: task1/JAVA/main.java
import java.util.Scanner;

public class main {

    // Функция проверяет, содержит ли подстрока все символы из T
    public static boolean allSymvl(String str, String t) {
        int[] count = new int[128];

        // Считаем символы в подстроке
        for (char c : str.toCharArray()) {
            count[c]++;
        }

        // Проверяем, хватает ли каждого символа из T
        for (char c : t.toCharArray()) {
            count[c]--;
            if (count[c] < 0) return false;
        }
        return true;
    }

    public static String minStr(String s, String t) {
        if (t.length() > s.length()) return "Не найдено";

        String result = "";
        int minLen = s.length() + 1;

        // Перебираем все возможные подстроки
        for (int i = 0; i < s.length(); i++) {
            for (int j = i; j < s.length(); j++) {
                String sub = s.substring(i, j + 1);

                if (allSymvl(sub, t)) {
                    if (sub.length() < minLen) {
                        minLen = sub.length();
                        result = sub;
                    }
                    break; // Для этого i нашли минимальную подстроку
                }
            }
        }

        return result.isEmpty() ? "Не найдено" : result;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        if (sc.hasNextLine()) {
            String s = sc.nextLine();
            if (sc.hasNextLine()) {
                String t = sc.nextLine();
                System.out.println(minStr(s, t));
            }
        }
        sc.close();
    }
}